llitgi/Coordinators/AppCoordinator.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  AppCoordinator.swift
+//  llitgi
+//
+//  Created by Xavi Moll on 31/07/2018.
+//  Copyright © 2018 xmollv. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SafariServices
+
+protocol Coordinator {
+    func start()
+}
+
+final class AppCoordinator: NSObject, Coordinator {
+    
+    //MARK: Private properties
+    private let splitViewController: UISplitViewController
+    private let tabBarController: UITabBarController
+    private let emptyViewController: EmptyDetailViewController
+    private let factory: ViewControllerFactory
+    private let userManager: UserManager
+    
+    private lazy var presentSafariClosure: ((SFSafariViewController) -> Void)? = { [weak self] sfs in
+        guard let strongSelf = self else { return }
+        sfs.delegate = strongSelf
+        strongSelf.splitViewController.showDetailViewController(sfs, sender: nil)
+    }
+    
+    //MARK: Lifecycle
+    init(window: UIWindow, tabBarController: UITabBarController = UITabBarController(), factory: ViewControllerFactory, userManager: UserManager) {
+        self.splitViewController = UISplitViewController()
+        self.tabBarController = tabBarController
+        self.emptyViewController = factory.instantiateEmptyDetail()
+        self.factory = factory
+        self.userManager = userManager
+        
+        super.init()
+        
+        let listViewController: ListViewController = self.factory.instantiateList(for: .myList)
```

Comment:
This code to configure the Lists depending on the type should be hidden elsewhere.